cJSON.c

```diff
@@ -935,7 +941,7 @@ static parse_buffer *buffer_skip_whitespace(parse_buffer * const buffer)
 /* Parse an object - create a new root, and populate. */
 CJSON_PUBLIC(cJSON *) cJSON_ParseWithOpts(const char *value, const char **return_parse_end, cJSON_bool require_null_terminated)
 {
-    parse_buffer buffer;
+    parse_buffer buffer = { 0 };
```

Comment:
I just checked. GCC is not optimizing out the `memset`. In case `{ 0 }` is invalid, replace it with `{ 0, 0, 0}`.